types.cc

```diff
@@ -314,9 +314,6 @@ struct string_type_impl : public concrete_type<sstring> {
     virtual bool is_byte_order_equal() const override {
         return true;
     }
-    virtual bool is_byte_order_comparable() const override {
-        return true;
-    }
     virtual bool is_string() const override {
```

Comment:
A lot of functions like this one can be converted to bool : 1 fields stored in the base class. Types are immutable so we don't need to recalculate this over and over.

Reply:
In fact, most of them can be a bitfield in reversed_type_impl since that is the only case where it is not known by just looking a _kind.

Can this be in a followup series?

Reply:
Yes of course. This one is long enough.